### PR TITLE
Update whisper.cc : Add logits to WhisperGenerationResult

### PR DESCRIPTION
**Pull request by sammygrey**
WhisperGenerationResult was missing return value for logits despite register_whisper having a readonly definition for said logits.
Logits are documented on https://opennmt.net/CTranslate2/python/ctranslate2.models.Whisper.html as being returned when return_logits_vocab is set to True, however, there was no written return value to access them within WhisperGenerationResult if it was set to True.

This fixes issue [#1779](https://github.com/OpenNMT/CTranslate2/issues/1779).

### CODE REVIEW

**Comment by MahmoudAshraf97** [comment]
also fixes #1822

**Comment by minhthuc2502** [comment]
Thank you for your fix. I will merge this
